### DISCUSSION TITLE
Callback on_change of TextField is not triggered when its value has been set programmatically

### DISCUSSION THREAD

**Discussion by RobertNeher** [flet-dev/flet#928] [Q&A]
Callback:
 ```
 def copyValue(e):
            goalField.value = e.control.value
            goalField.update()
```
Question:
Setting the value of a TextField programmatically (see code above), does not fire the event/callback on_change in target TextField
Is this by intent?

**Comment by ndonkoHenri** [comment]
That's normal, I think.
To resolve your problem, you have to call your `on_change` callback, and pass `None` as the parameter;
```python
def copyValue(e):
       goalField.value = e.control.value
       goalField.update()
       your_on_change(None)
``` 
Let me know if it resolved your issue.
